vllm_spyre/platform.py

```diff
@@ -515,3 +523,216 @@ def get_max_output_tokens(self, prompt_len: int) -> int:
                 max_new_tokens = max(max_new_tokens, shape['new_tokens'])
 
         return max_new_tokens
+
+    @classmethod
+    def _handle_disable_compilation(cls, vllm_config: VllmConfig,
+                                    is_decoder: bool):
+        """
+        For decoder models we want to respect the `REQUIRE_PRECOMPILED_DECODERS`
+        environment variable, which disallows torch_sendnn from compiling new
+        graphs and only allows it to load pre-compiled graphs.
+        In order to do this, we must load up some config from the torch_sendnn
+        cache and check to make sure that the current vllm config matches,
+        otherwise the cached artifacts cannot be used.
+
+        For encoder models, we do not allow disabling compilation.
+        """
+        req_precompiled_decoder_env_var = "REQUIRE_PRECOMPILED_DECODERS"
+
+        disable_flag = os.getenv(req_precompiled_decoder_env_var, "").lower()
+        if disable_flag not in ("1", "true"):
+            return
+
+        # If this isn't a decoder model, re-enable compilation
+        if not is_decoder:
+            logger.info("Unsetting %s because %s is not a decoder model",
+                        DISABLE_COMPILATION_ENV_VAR,
+                        vllm_config.model_config.model)
+            os.environ[DISABLE_COMPILATION_ENV_VAR] = "false"
+            return
+
```

Comment:
A little tweak to the logic here after changing to `REQUIRE_PRECOMPILED_DECODERS`

```suggestion
      if not is_decoder:
          return

      # REQUIRE_PRECOMPILED_DECODERS forces compilation to be disabled
      disable_compilation = os.getenv(DISABLE_COMPILATION_ENV_VAR, "true")
      if disable_compilation != "true"):
          raise ValueError("DISABLE_COMPILATION is set to allow compilation "
              "in conflict with REQUIRE_PRECOMPILED_DECODERS which disables it.",
          )

```